doc/source/skips/3-transition-to-v1.rst

```diff
@@ -0,0 +1,243 @@
+  easier for new users to find out which functions they should try out for a
+  specific task.
+
+The question is, how do we make this transition while causing as little
+disruption as possible?
+
+This document proposes releasing 0.19 as the final 0.x series release, then
+immediately releasing a nearly identical 0.20 release that warns users about
+breaking changes in 1.0, thus giving them an opportunity to pin their
+scikit-image dependency to 0.19.x. The warning would also point users to a
+transition guide to prepare their code for 1.0. See `Implementation` for
+details.
+
+This approach ensures that all users get ample warning, and a chance to ensure
+that their scripts and libraries will continue to work after 1.0 is released.
+Users who don't have the time or inclination to make the transition will be
+able to pin their dependencies correctly. Those who prefer to be on the cutting
+edge will also be able to plan around the 1.0 release and update their code
+correctly, in sync with scikit-image.
+
+Related Work
+------------
+
+`pandas` released 1.0.0 in January 2020, including many backwards-incompatible
+API changes [3]_. `scipy` released version 1.0 in 2017, but, given its stage of
+maturity and position at the base of the scientific Python ecosystem, opted not
+to make major breaking changes [4]_. However, SciPy has adopted a policy of
+adding upper-bounds on dependencies [5]_, acknowledging that the ecosystem as a
+whole makes backwards incompatible changes on a 2 version deprecation cycle.
+
+Implementation
+--------------
+
+The details of the proposal are as follows:
+
+- scikit-image 0.19 will be the final *true* 0.x release. It contains some new
+  features, bug fixes, and several API changes following on from deprecations
+  in 0.17.
+- shortly after 0.19, we release 0.20, which is identical except that it emits
+  a warning at import time. The warning reads something like the following:
+  "scikit-image 1.0 will be released later this year and will contain breaking
+  changes. To ensure your code keeps running, please install
+  ``scikit-image<=0.19.*``. To silence this warning but still depend on
+  scikit-image after 1.0 is released, install ``scikit-image!=0.20.*``." The
+  warning also contains a link for further details, and instructions for
+  managing the dependency in both conda and pip environments.
+- After 0.20, we make all the API changes we need, without deprecation cycles.
+  Importantly, for every API change, we add a line to a "scikit-image 1.0
+  transition guide" in the documentation, which maps every changed
+  functionality in the library from its old form to its new form. These changes
+  are tracked on a GitHub issue [7]_ and in the 1.0 milestone [8]_.
+- Once the transition has happened in the repository, we release 1.0.0a0, an
+  alpha release which contains a global warning pointing to the transition
+  guide, as well as all of the new functionality. We also release 0.21, which
```

Comment:
```suggestion
  guide, as well as all of the new functionality. We also release 0.20, which
```
0.20 here, right?

Reply:
As a related question, we already have a handful of deprecation warnings that mention a release "0.20" by name. Should we just update those messages to say "1.0" instead prior to releasing 0.19? To follow this skip we should not be introducing any new deprecations in 0.20.

Places that I could find mention of release "0.20" by name currently are:
`skimage.viewer` module
`scripts/skivi.py`
`io/_plugins/qt_plugin.py`
`skimage/measure/_find_contours.py`
`feature/peak.py`
`feature/corner.py`

Reply:
@grlee77 actually, something we maybe didn't mention in the meeting: my idea was that we would release 0.21 as a "0.20 redux" once the transition guide is complete. This will allow people who are on `!=0.20` the chance to know when they can *actually* migrate their library. Before that release, 0.20 can warn them that this is coming, but the guide won't contain anything, so they have to... Poll the website for updates? It's not ideal.

So yeah, I meant 0.21 here.

Regarding deprecations, I was also going to make a separate issue: I think 0.19 should contain *no* deprecations — assuming we move forward on the SKIP, of course. We'll have a better sense of that this week, I hope. Anyway, it's the final 0.x version, so the 0.x API could be "frozen" and people will fall into only two camps: people who depend on `<=0.19`, and they shouldn't have to worry about deprecations, and people who will move to 1.0, and they shouldn't have to first deal with deprecations only to have to do a full migration later.

Does that make sense? It only occurred to me as I was writing the SKIP, so we haven't really discussed it so far.